source/MRMesh/MRVectorTraits.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include "MRMesh/MRMeshFwd.h"
+
+#include <cassert>
+
+namespace MR
+{
+
+// Common traits for (mathematical) vectors.
+
+template <typename T>
+struct VectorTraits
+{
+    // The base template handles scalars (or just non-vectors).
+
+    using BaseType = T;
+    static constexpr int size = 1;
+
+    // Currently this doesn't forward the value, for simplicity. (In all specializations.)
+    // For scalars this intentionally doesn't check the index.
+    template <typename U>
+    [[nodiscard]] static constexpr auto&& getElem( int i, U&& value ) { (void)i; return value; }
+};
+
+template <typename T>
+struct VectorTraits<Vector2<T>>
+{
+    using BaseType = T;
+    static constexpr int size = 2;
```

Comment:
Vector2<T> already have ::elements (other vectors too)

Reply:
I still need the traits to be able to accept ImVec2/4 and for scalars (where BaseType=T and size=1)